### DISCUSSION TITLE
Allocate specific UID for package.

### DISCUSSION THREAD

**Discussion by CRKatri** [ProcursusTeam/Procursus#581] [Q&A] [answered]
I'm fixing my bsdgames PR, and it needs a `games` user and group, for iOS I have 
```bash
#!/bin/sh

if [ "$1" = 'configure' ]; then
	if !getent group 600 | grep -q games; then
		# group 600 is not `games`! Time to error out
		echo "Group 600 is already allocated but not games, this isn't good"
		exit 1
	else
		pw group add -n games -u 600 &>/dev/null
		pw user add -n games -u 600 -g 600 -s /usr/sbin/nologin -d /var/empty &>/dev/null
	fi
fi
```
but I don't know what to do for the MacOS one, I need UID and GID to be 600, because multiple games and files have to be packaged while owned by `games`. On iOS UIDs and GIDs 300-999 are available for packages to take, but I'm not sure if that's the case for MacOS.

**Comment by Diatrus** [comment]
Why do you need 600?

**Reply by CRKatri** [reply]
It doesn't have to be 600, but I need a static one, cause some games are `setgid`d to write to their score file, and `/var/games` needs to be owned by `root:games`

**Comment by CRKatri** [comment, marked as answer]
I’m closing this as I am going to `chown` and `chmod` the files in the `postinst`, I am also going to add a flag to `pw` to allocate a `UID`/`GID` in the 300-999 range instead of 1000-?, this will be used for this `games` user and group and the `_apt` user and such.

**Reply by Diatrus** [reply]
Please see how the Redis postscripts do it. There is a dpkg function specifically for stuff like this other than chmod.

**Reply by CRKatri** [reply]
You mean with `dpkg-statoverride`?
